haystack/preview/dataclasses/document.py

```diff
@@ -113,6 +115,10 @@ def __post_init__(self):
         """
         Generate the ID based on the init parameters.
         """
+        # if id_hash_keys is None or empty, use the default factory
+        if not self.id_hash_keys:
+            self.id_hash_keys = id_hash_keys_default_factory()
```

Comment:
I haven't checked yet, but have you tried to access the default factory of the field instead of using an external function? Not like it makes a difference, it would only make it slightly more concise

Reply:
I initially tried, but `fields` is a tuple.
So my code would have been similar to this in `__post_init__`:
```python
# from dataclasses import fields

id_hash_keys_default_factory = [f for f in fields(self) if f.name=="id_hash_keys"][0].default_factory
```

It seemed convoluted and not very clear to me.
So I discarded this approach.